toolkit/tools/scheduler/schedulerutils/implicitprovides.go

```diff
@@ -60,7 +60,8 @@ func replaceNodesWithProvides(res *BuildResult, pkgGraph *pkggraph.PkgGraph, pro
 	// Make this node the parent node for the new implicit provide node.
 	// - By making a run node the parent node, it will inherit the identical runtime dependencies of the already setup node.
 	for _, node := range pkgGraph.AllRunNodes() {
-		if rpmFileProviding == node.RpmPath {
+		// We need to filter out any non-run nodes since cached remote nodes are considered "Run" nodes for the graph.
```

Comment:
Does that mean the term "run node" is overloaded and means different things from the point of the view of the graph and the node itself? And will other places calling `AllRunNodes` (and maybe also `AllBuildNodes`) still get the nodes they expect? In other words: I'm curious, if it would make sense to move the filtering into the the `All*Nodes` methods?